regression-test/suites/export/test_array_export.groovy

```diff
@@ -0,0 +1,132 @@
+
+        result1 = sql """
+            CREATE TABLE IF NOT EXISTS ${tableName} (
+              `k1` INT(11) NULL COMMENT "",
+              `k2` ARRAY<SMALLINT> NOT NULL COMMENT "",
+              `k3` ARRAY<INT(11)> NOT NULL COMMENT "",
+              `k4` ARRAY<BIGINT> NOT NULL COMMENT "",
+              `k5` ARRAY<CHAR> NOT NULL COMMENT "",
+              `k6` ARRAY<VARCHAR(20)> NULL COMMENT "",
+              `k7` ARRAY<DATE> NOT NULL COMMENT "", 
+              `k8` ARRAY<DATETIME> NOT NULL COMMENT "",
+              `k9` ARRAY<FLOAT> NOT NULL COMMENT "",
+              `k10` ARRAY<DOUBLE> NOT NULL COMMENT "",
+              `k11` ARRAY<DECIMAL(27, 9)> NULL COMMENT ""
+            ) ENGINE=OLAP
+            DUPLICATE KEY(`k1`)
+            DISTRIBUTED BY HASH(`k1`) BUCKETS 1
+            PROPERTIES (
+            "replication_allocation" = "tag.location.default: 1",
+            "storage_format" = "V2"
+            )
+        """
+
+        // DDL/DML return 1 row and 3 column, the only value is update row count
+        assertTrue(result1.size() == 1)
+        assertTrue(result1[0].size() == 1)
+        assertTrue(result1[0][0] == 0, "Create table should update 0 rows")
+
+        sql """ INSERT INTO ${tableName} VALUES
+                        (1, [1, 2, 3], [32767, 32768, 32769], [65534, 65535, 65536], ['a', 'b', 'c'], ["hello", "world"], 
+                        ['2022-07-13'], ['2022-08-15 12:30:00'], [0.33, 0.67], [3.1415926, 0.878787878], [4, 5.5, 6.67])
+                        """
+        
+        sql """ INSERT INTO ${tableName} VALUES
+                        (2, [4, 5, 6], [32767, 32768, 32769], [65534, 65535, 65536], ['d', 'e', 'f'], ["good", "luck"], 
+                        ['2022-07-13'], ['2022-08-15 15:59:59'], [0.333336, 0.666677], [3.1415926, 0.878787878], [4, 5.555, 6.66667])
+                        """
+        
+        qt_select_default """ SELECT * FROM ${tableName} t ORDER BY k1; """
+
+        // check outfile
+        File path = new File(outFilePath)
+        if (!path.exists()) {
+            assert path.mkdirs()
+        } else {
+            throw new IllegalStateException("""${outFilePath} already exists! """)
+        }
+        sql """
+            SELECT * FROM ${tableName} t ORDER BY k1 INTO OUTFILE "file://${outFilePath}/";
+        """
+        File[] files = path.listFiles()
+        assert files.length == 1
+        List<String> outLines = Files.readAllLines(Paths.get(files[0].getAbsolutePath()), StandardCharsets.UTF_8);
+        List<String> baseLines = Files.readAllLines(Paths.get("""${context.config.dataPath}/export/test_array_export.out"""), StandardCharsets.UTF_8)
+        for (int j = 0; j < outLines.size(); j ++) {
+            String[] outLine = outLines.get(j).split("\t")
+            String[] baseLine = baseLines.get(j + 2).split("\t")
+            for (int slotId = 0; slotId < outLine.size(); slotId ++) {
+                assert outLine[slotId] == baseLine[slotId]
```

Comment:
P0 regression failed here caused by format and precision issues when reading files

Reply:
ok，I will re-check.